sound/soc/sof/debug.c

```diff
@@ -38,9 +39,20 @@ static ssize_t sof_dfsentry_read(struct file *file, char __user *buffer,
 	if (count > size - pos)
 		count = size - pos;
 
+	/* align io read start to u32 multiple */
+	pos = ALIGN_DOWN(pos, 4);
+
 	/* intermediate buffer size must be u32 multiple */
 	size = ALIGN(count, 4);
 
+	/* if start position is unaligned, read extra u32 */
+	if (unlikely(pos != *ppos)) {
+		size += 4;
+		skip = *ppos - pos;
+		if (pos + size > dfse->size)
+			size -= 4;
```

Comment:
could we instead only increment size when possible, e.g

````c
if (unlikely(pos != *ppos)) {
		skip = *ppos - pos;
		if (pos + size +4  <= dfse->size)
			size += 4;
````

the +4 then -4 looks odd.

Reply:
Ack. Yes, this is better, fixed and reuploaded.